src/main/java/appeng/items/tools/powered/ToolWirelessTerminal.java

```diff
@@ -84,26 +155,162 @@ public void addCheckedInformation( final ItemStack stack, final EntityPlayer pla
 			{
 				final String encKey = tag.getString( "encryptionKey" );
 
-				if( encKey == null || encKey.isEmpty() )
+				return !( encKey == null || encKey.isEmpty() );
+			}
+		}
+		return false;
+	}
+
+	public boolean getIsUsable( final ItemStack stack, Entity holder )
+	{
+		if( stack.hasTagCompound() )
+		{
+			final NBTTagCompound tag = Platform.openNbtData( stack );
+			if( tag != null )
+			{
+				boolean usable = ( this.hasPower( null, 0.5, stack ) && ToolWirelessTerminal.isLinked( stack ) );
+				if( usable )
 				{
-					lines.add( GuiText.Unlinked.getLocal() );
+					final long parsedKey = Long.parseLong( this.getEncryptionKey( stack ) );
+					final ILocatable securityStation = AEApi.instance().registries().locatable().getLocatableBy( parsedKey );
+					if( securityStation == null )
+					{
+						usable = false;
+					}
 				}
-				else
+				if( usable )
 				{
-					lines.add( GuiText.Linked.getLocal() );
+					usable = ((ToolWirelessTerminal)stack.getItem()).rangeCheck( stack, holder );
 				}
+				return usable;
 			}
 		}
-		else
+
+		return false;
+	}
+
+	private void updateIsUsable( final ItemStack stack, final Entity entity )
+	{
+		if( stack.hasTagCompound() )
+		{
+			final NBTTagCompound tag = Platform.openNbtData( stack );
+			if( tag != null )
+			{
+				boolean usable = ( this.hasPower( null, 0.5, stack ) && ToolWirelessTerminal.isLinked( stack ) );
+				if( usable )
+				{
+					final long parsedKey = Long.parseLong( this.getEncryptionKey( stack ) );
+					final ILocatable securityStation = AEApi.instance().registries().locatable().getLocatableBy( parsedKey );
+					if( securityStation == null )
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)